crates/burn-tensor/src/tensor/api/kind.rs

```diff
@@ -31,10 +31,25 @@ impl<B: Backend> TensorPrimitive<B> {
     }
 }
 
+impl<B: Backend> Primitive for TensorPrimitive<B> {
+    fn dtype(&self) -> DType {
+        match self {
+            TensorPrimitive::Float(tensor) => tensor.dtype(),
+            TensorPrimitive::QFloat(tensor) => tensor.dtype(),
+        }
+    }
+}
+
+/// Primitive trait for tensor.
+pub trait Primitive: Clone + Send + Sync + core::fmt::Debug {
+    /// The dtype of the tensor.
+    fn dtype(&self) -> DType;
+}
```

Comment:
Maybe it would be a good idea to put all tensor metadata here, so to include shape.